README.md

```diff
@@ -1,35 +1,60 @@
 
-use `yarn size` to check contract size
+```bash
+curl -L https://foundry.paradigm.xyz | sh
+```
 
-use `yarn doc` to generate natspec docs
+You can download and install dependencies with:
 
-use `yarn lint` to lint the code
+```bash
+forge install
+```
 
-use `yarn tree` to generate a Solidity dependency tree
+If you run into trouble with `forge install`, try using `git submodule update --init --recursive` to ensure that nested submodules have been properly initialized.
 
-<br>
+Some useful commands:
 
-use `yarn deploy:mainnet` and `yarn deploy:goerli` to deploy and verify (see .env.example for required env vars, using a ledger by default).
+| Command               | Description                                         |
+| --------------------- | --------------------------------------------------- |
+| `forge install`       | Install the dependencies.                           |
+| `forge build`         | Compile the contracts and write artifacts to `out`. |
+| `forge fmt`           | Lint.                                               |
+| `forge test`          | Run the tests.                                      |
+| `forge build --sizes` | Get contract sizes.                                 |
+| `forge coverage`      | Generate a test coverage report.                    |
+| `foundryup`           | Update foundry. Run this periodically.              |
+| `forge clean`         | Remove the build artifacts and cache directories.   |
 
-## Code coverage
-Run `yarn coverage`to display code coverage summary and generate an LCOV report
+To learn more, visit the [Foundry Book](https://book.getfoundry.sh/) docs.
 
-To display code coverage in VSCode:
-- You need to install the [coverage gutters extension (Ryan Luker)](https://marketplace.visualstudio.com/items?itemName=ryanluker.vscode-coverage-gutters) or any other extension handling LCOV reports
-- ctrl shift p > "Coverage Gutters: Display Coverage" (coverage are the colored markdown lines in the left gutter, after the line numbers)
+We recommend using [Juan Blanco's solidity extension](https://marketplace.visualstudio.com/items?itemName=JuanBlanco.solidity) for VSCode.
 
-## PR
-Github CI flow will run both unit and forked tests, log the contracts size (with the tests) and check linting compliance.
\ No newline at end of file
+## Utilities
+
+For convenience, several utility commands are available in `util.sh`. To see a list, run:
+
+```bash
+`bash util.sh --help`.
+```
+
+Or make the script executable and run:
+
+```bash
+./util.sh --help
+```
+
+## Terminals
+
+Juicebox projects can accept funds through one or more *terminals*, which can manage both inflows (via payments) and outflows (via redemptions). Terminals usually only accept one token for payments, but if a project has a swap terminal, the swap terminal can accept any token and swap it for tokens which the project *can* accept. After swapping, it redirects the payment to the primary terminal for the token it received.
```

Comment:
Maybe add emphasis on the diff classic (pay/redeem) terminal vs this one (pay only)?